File: scripts/R/test_correct.R
source("util/setup.R")

dir = file.path(plot_dir,paste("run",Sys.Date(),sep="_"))
dir.exists(dir) || dir.create(dir)

df = read.csv(file.path(cache_dir,"test_correct.csv"))
df = df %>% group_by(sid,condition,speaker) %>%
    mutate(trial = 1:length(corr)) %>%
    mutate(test_correct = test_correct == "true")

df %>% group_by(sid) %>% summarize(accuracy = mean(test_correct))

ggplot(df,aes(x=speaker,y=corr,color=test_correct)) +
    geom_point(position=position_jitterdodge(jitter.width=0.2,dodge.width=0.3),
        alpha=0.5,size=1,aes(shape=test_correct)) +
    stat_summary(geom="pointrange",fun.data="mean_cl_boot",size=0.3,
        position=position_nudge(x=0.4)) +
    geom_abline(slope=0,intercept=0,linetype=2) +
    scale_color_brewer(palette='Set1') +
    coord_cartesian(xlim=c(0.5,4.5)) +
    facet_grid(condition~sid)

ggsave(file.path(dir,"correct_corr_vs_control.pdf"),width=9,height=7)

dfcor = df %>%
    group_by(sid,condition,trial,test_correct) %>%
    spread(speaker,corr)

ggplot(dfcor,aes(x=pmax(fem1,fem2),y=male)) +
    geom_point(aes(color=test_correct)) +
    geom_abline(intercept=0,slope=1) + facet_grid(condition~sid) +
    scale_color_brewer(palette='Set1')
# ggsave(file.path(dir,"correct_corr_male_v_female_by_condition.pdf"),
#     width=9,height=5)

ggplot(dfcor,aes(x=pmax(fem1,fem2),y=male)) +
    geom_point(aes(color=test_correct)) +
    geom_abline(intercept=0,slope=1) + facet_grid(~sid) +
    scale_color_brewer(palette='Set1')
# ggsave(file.path(dir,"correct_corr_male_v_female.pdf"),width=9,height=2.5)

File: scripts/julia/test_correct.jl
using DrWatson; quickactivate(@__DIR__,"german_track")
using GermanTrack

stim_info = JSON.parsefile(joinpath(stimulus_dir(),"config.json"))
eeg_files = filter(x -> occursin(r"_mcca34\.mcca_proj$",x),readdir(data_dir()))
eeg_files = eeg_files[1:1]

# TODO: we don't need this file format, we can use the 65 components directly,
# to reduce memory load.

fbounds = trunc.(Int,round.(exp.(range(log(90),log(3700),length=5))[2:end-1],
    digits=-1))

encoding = JointEncoding(PitchSurpriseEncoding(),ASBins(fbounds))

correct = [
    string("correct",cond) =>
        @λ(_row.condition == cond && _row.correct ? all_indices : no_indices)
    for cond in ["feature","object","test"]
]
alltrials = [
    string("all",cond) => @λ(_.condition == cond ? all_indices : no_indices)
    for cond in ["feature","object","test"]
]

df = train_test(
    StaticMethod(NormL2(0.2),cor),
    SpeakerStimMethod(encoding=encoding),
    resample = 64,
    eeg_files,stim_info,
    train = correct,
    test = alltrials,
    skip_bad_trials = true,
)
alert()


dir = joinpath(plotsdir(),string("results_",Date(now())))
isdir(dir) || mkdir(dir)
df.condition = replace.(df.condition,Ref(r"traincorrect([[:alnum:]]+)_.*" => s"\1"))

R"""

library(dplyr)
library(ggplot2)

df = $df %>% group_by(sid,condition,source) %>%
    mutate(trial = 1:length(value))

df %>% group_by(sid) %>% summarize(accuracy = mean(test_correct))

ggplot(df,aes(x=source,y=value,color=test_correct)) +
    geom_point(position=position_jitter(width=0.1),
        alpha=0.5,size=0.8) +
    stat_summary(geom="pointrange",fun.data="mean_cl_boot",size=0.2,
        position=position_nudge(x=0.3)) +
    geom_abline(slope=0,intercept=0,linetype=2) +
    scale_color_brewer(palette='Set1') +
    coord_cartesian(xlim=c(0.5,5.5)) +
    facet_grid(condition~sid)

ggsave(file.path($dir,"test_correct.pdf"),width=9,height=7)

"""

